src/Carousel.vue

```diff
@@ -66,6 +66,12 @@ export default {
     };
   },
   mixins: [autoplay],
+  // use `provide` to avoid `Slide` being nested with other components
+  provide() {
```

Comment:
Woah this is cool, I've never seen it before. Could lead to some really good refactoring for `v1.0.0` 🎉

Reply:
Haha, this is new in 2.2.0+.
And I've noticed that in `Navigation` and `Pagination`, you're using
``` js
{
  /**
   * link on Carousel
   */
  parentContainer: this.$parent
}
```
This is ok in vue-carousel, since these two components are directly nested in `Carousel`. But when the DOM structure has been modified, it likely leads to some errors. So I highly recommend that we use `provide/inject` to link to the `Carousel` instance.
What do you think?😄

Reply:
Yeah I think it's much better! Good fix, I'm just going to test the branch now and will merge later today